infoschema/tables_test.go

```diff
@@ -871,17 +873,51 @@ func (s *testTableSuite) TestForTableTiFlashReplica(c *C) {
 
 func (s *testClusterTableSuite) TestForClusterServerInfo(c *C) {
 	tk := testkit.NewTestKit(c, s.store)
-	mockAddr := "127.0.0.1:10080"
 	instances := []string{
-		"pd,127.0.0.1:11080," + mockAddr + ",mock-version,mock-githash",
-		"tidb,127.0.0.1:11080," + mockAddr + ",mock-version,mock-githash",
-		"tikv,127.0.0.1:11080," + mockAddr + ",mock-version,mock-githash",
+		"pd,127.0.0.1:11080," + s.listenAddr + ",mock-version,mock-githash",
+		"tidb,127.0.0.1:11080," + s.listenAddr + ",mock-version,mock-githash",
+		"tikv,127.0.0.1:11080," + s.listenAddr + ",mock-version,mock-githash",
 	}
 	fpExpr := `return("` + strings.Join(instances, ";") + `")`
 	c.Assert(failpoint.Enable("github.com/pingcap/tidb/infoschema/mockClusterInfo", fpExpr), IsNil)
 	defer func() { c.Assert(failpoint.Disable("github.com/pingcap/tidb/infoschema/mockClusterInfo"), IsNil) }()
 
+	checkInfoRows := func(re *testkit.Result) {
+		rows := re.Rows()
+		c.Assert(len(rows), Greater, 0)
+
+		// Currently only TiDB implement this.
+		// TODO: fix me after tikv/pd server support this.
+		typeMap := map[string]struct{}{
+			"tidb": {},
+		}
+		addrMap := map[string]struct{}{
+			s.listenAddr: {},
+		}
+		nameMap := map[string]struct{}{
+			"cpu":  {},
+			"mem":  {},
+			"net":  {},
+			"disk": {},
+		}
+		for _, row := range rows {
+			tp := row[0].(string)
+			addr := row[1].(string)
+			name := row[2].(string)
+			delete(typeMap, tp)
+			delete(addrMap, addr)
+			delete(nameMap, name)
+		}
+		c.Assert(len(typeMap), Equals, 0)
+		c.Assert(len(addrMap), Equals, 0)
+		c.Assert(len(nameMap), Equals, 0)
+	}
+
 	re := tk.MustQuery("select * from information_schema.CLUSTER_LOAD;")
+	checkInfoRows(re)
+	re = tk.MustQuery("select * from information_schema.CLUSTER_HARDWARE;")
+	checkInfoRows(re)
+	re = tk.MustQuery("select * from information_schema.CLUSTER_SYSTEMINFO;")
```

Comment:
The tests are too weak, please add more tests for them

Reply:
em...Any succession? since the system info of every machine was different.

Reply:
@crazycs520 Maybe you can reference the https://github.com/tikv/tikv/pull/6135 unit tests.

Reply:
done.